services/wallet/concurrent.go

```diff
@@ -45,17 +45,56 @@ type TransferDownloader interface {
 	GetTransfersByNumber(context.Context, *big.Int) ([]Transfer, error)
 }
 
+var balancesCache = map[common.Address]map[*big.Int]*big.Int{}
```

Comment:
```suggestion
var (
    balancesCache = make(map[common.Address]map[*big.Int]*big.Int)
    balancesCacheLock sync.RWMutex
)
```

Is it ok if this cache is not persistent?

Another approach would be that this cache belongs to something that calls `downloadEthConcurrently`. It can then assign the reference to the cache to `ConcurrentDownloader`.

Reply:
> Is it ok if this cache is not persistent?
yes it is ok, this is to prevent requesting the balance from the same block during initial scan, not like we want to persist the whole history of these calls